model/workflow.go

```diff
@@ -508,11 +508,13 @@ type End struct {
 
 // UnmarshalJSON ...
 func (e *End) UnmarshalJSON(data []byte) error {
-
 	endMap := make(map[string]json.RawMessage)
 	if err := json.Unmarshal(data, &endMap); err != nil {
-		e.Terminate = false
-		e.Compensate = false
+		e.Terminate, err = unmarshalBool(data)
```

Comment:
So the `data` here contains just `false` or `true`? If there's an error in the first conditional and we enter here, it means that the `End` structure is wrong?

Reply:
I like the second better. And we can reuse this func in other scenarios. Nice! Thank u!